src/core/tests/preprocess.cpp

```diff
@@ -209,6 +212,129 @@ TEST(pre_post_process, tensor_element_type_and_scale) {
+    EXPECT_EQ(f->get_parameters().size(), 1);
+    EXPECT_EQ(f->get_parameters().front()->get_layout(), "NHWC");
+    EXPECT_EQ(ov::layout::get_layout(f->input(0)), "NHWC");
+    EXPECT_EQ(f->get_parameters().front()->get_friendly_name(), name);
+    EXPECT_EQ(f->get_parameters().front()->get_output_tensor(0).get_names(), tensor_names);
+    EXPECT_EQ(f->get_parameters().front()->get_partial_shape(), (PartialShape{Dimension::dynamic(), 2, 2, 3}));
+    EXPECT_EQ(f->get_result()->get_output_partial_shape(0), (PartialShape{Dimension::dynamic(), 2, 2, 1}));
+}
+
+TEST_P(pre_post_process_to_gray, convert_color_nchw_to_nchw) {
+    auto COLOR_FORMAT = GetParam();
+
+    auto f = create_simple_function(element::f32, PartialShape{Dimension::dynamic(), 1, 2, 2});
+    auto p = PrePostProcessor(f);
+    p.input().tensor().set_layout("NCHW").set_color_format(COLOR_FORMAT);
+    p.input().model().set_layout("NCHW");
+    p.input().preprocess().convert_color(ColorFormat::GRAY);
+    f = p.build();
+
+    EXPECT_EQ(f->get_parameters().front()->get_layout(), "NCHW");
+    EXPECT_EQ(f->get_parameters().front()->get_partial_shape(), (PartialShape{Dimension::dynamic(), 3, 2, 2}));
+    EXPECT_EQ(f->get_result()->get_output_partial_shape(0), (PartialShape{Dimension::dynamic(), 1, 2, 2}));
+}
+
+TEST_P(pre_post_process_to_gray, convert_color_nhwc_to_nchw) {
+    auto COLOR_FORMAT = GetParam();
+
+    auto f = create_simple_function(element::f32, PartialShape{Dimension::dynamic(), 1, 2, 2});
+    auto p = PrePostProcessor(f);
+    p.input().tensor().set_layout("NHWC").set_color_format(COLOR_FORMAT);
+    p.input().model().set_layout("NCHW");
+    p.input().preprocess().convert_color(ColorFormat::GRAY);
+    f = p.build();
+
+    EXPECT_EQ(f->get_parameters().front()->get_layout(), "NHWC");
+    EXPECT_EQ(f->get_parameters().front()->get_partial_shape(), (PartialShape{Dimension::dynamic(), 2, 2, 3}));
+    EXPECT_EQ(f->get_result()->get_output_partial_shape(0), (PartialShape{Dimension::dynamic(), 1, 2, 2}));
+}
+
+TEST_P(pre_post_process_to_gray, convert_color_nchw_to_nhwc) {
+    auto COLOR_FORMAT = GetParam();
+
+    auto f = create_simple_function(element::f32, PartialShape{Dimension::dynamic(), 2, 2, 1});
+    auto p = PrePostProcessor(f);
+    p.input().tensor().set_layout("NCHW").set_color_format(COLOR_FORMAT);
+    p.input().model().set_layout("NHWC");
+    p.input().preprocess().convert_color(ColorFormat::GRAY);
+    f = p.build();
+
+    EXPECT_EQ(f->get_parameters().front()->get_layout(), "NCHW");
+    EXPECT_EQ(f->get_parameters().front()->get_partial_shape(), (PartialShape{Dimension::dynamic(), 3, 2, 2}));
+    EXPECT_EQ(f->get_result()->get_output_partial_shape(0), (PartialShape{Dimension::dynamic(), 2, 2, 1}));
+}
+
+TEST_P(pre_post_process_to_gray, convert_color_asserts) {
+    auto COLOR_FORMAT = GetParam();
+
+    // No "N" dimension name
+    OV_EXPECT_THROW(auto f = create_simple_function(element::f32, PartialShape{Dimension::dynamic(), 2, 2, 1});
```

Comment:
It looks like these test can be converted to multiple cases and be parameterized where `color format`, `simple model shape`,  and `layout` could be test parameters. Consider use another fixture which use tuple of parameters and make this test simpler 

or maybe refactor each test case to
```cpp
{
    auto f = create_simple_function(element::f32, PartialShape{Dimension::dynamic(), 2, 2, 1});
    auto p = PrePostProcessor(f);
    p.input().tensor().set_layout("DHWC").set_color_format(COLOR_FORMAT);
    p.input().preprocess().convert_color(ColorFormat::GRAY);
    // To point what action throws exception
    OV_EXPECT_THROW(p.build(), ov::AssertFailure, ::testing::HasSubstr("Dimension name 'N' is not found in layout"));
}
```